aurora-rust-sdk/aurora-sdk-integration-tests/src/aurora_engine/repo.rs

```diff
@@ -199,6 +200,23 @@ async fn add_wasm_target(engine_path: &Path, toolchain: &str) -> anyhow::Result<
     Ok(())
 }
 
+/// Recursively moves up the path tree, starting with the current directory, to find a `target`
+/// directroy.
+fn find_target_dir() -> anyhow::Result<PathBuf> {
+    let pwd = Path::new(".").canonicalize()?;
+    let mut current_base = pwd.as_path();
+    let mut result = current_base.join(TARGET);
+    while !result.exists() {
+        if let Some(p) = current_base.parent() {
+            current_base = p;
+        } else {
+            return Err(anyhow::Error::msg("Failed to find target directory"));
+        }
+        result = current_base.join(TARGET);
+    }
+    Ok(result)
+}
+
```

Comment:
I would add a unit test for this function like this:
```rust
#[test]
fn test_find_target_dir() {
    let result = find_target_dir().unwrap();
    assert_eq!(result, Path::new("../").canonicalize().unwrap().join(TARGET));
}
```
I suppose in the moment of running this test, the directory `target` should exist.

Reply:
Added in https://github.com/aurora-is-near/aurora-contracts-sdk/pull/8/commits/3db3859ad0e4f7c3d418a7285f66f4320c2231a9